File: level2/ex39.js
function solution(skillOrder, skillTrees) {
  let answer = 0;
  
  // 문자열 => 배열
  skillOrder = skillOrder.split('');
  
  for (const skillTree of skillTrees) {
      // 스킬 위치
      const index = skillOrder.map(value => skillTree.indexOf(value))
                              .map(value => value === -1 ? 26 : value);
      
      // 가능 여부
      let possible = true;
      
      // 순서 확인 
      for (let i = 0; i < index.length; i++) {
          // 잘못된 순서인 경우
          if (index[i] > index[i + 1]) {
              possible = false;
              break;
          }
      }

      // 가능한 순서인 경우
      if (possible) {
          answer++;
      }
  }
  
  return answer;
}

File: level2/ex39.py
def solution(skill, skill_trees):
    answer = 0
    
    for skill_tree in skill_trees:
        # 스킬 위치
        index = [skill_tree.find(value) for value in skill]

        # 필요없는 위치 제거
        for _ in range(len(index)):
            if index[-1] == -1:
                index = index[:-1]

        # 가능한 순서인 경우
        if sorted(index) == index and -1 not in index:
            answer += 1
        
    return answer